packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -62,6 +69,47 @@ export type TreeNodeSchema<
 			TCustomMetadata
 	  >;
 
+/**
+ * Stores annotations for an individual allowed type.
+ * @alpha
+ */
+export interface AnnotatedAllowedType<T = LazyItem<TreeNodeSchema>> {
+	/**
+	 * Annotations for the allowed type.
+	 */
+	readonly metadata: AllowedTypeMetadata;
+	/**
+	 * The allowed type the annotations apply to in a particular schema.
+	 */
+	readonly type: T;
+}
+
+/**
+ * Stores annotations for a set of evaluated annotated allowed types.
+ * @alpha
+ */
+export interface NormalizedAnnotatedAllowedTypes {
+	/**
+	 * Annotations that apply to a set of allowed types.
+	 */
+	readonly metadata: AllowedTypesMetadata;
+	/**
+	 * All the evaluated allowed types that the annotations apply to. The types themselves are also individually annotated.
+	 */
+	readonly types: readonly AnnotatedAllowedType<TreeNodeSchema>[];
+}
+
+/**
+ * Checks if the input is an {@link AnnotatedAllowedTypes}.
+ */
+export function isAnnotatedAllowedTypes(
+	allowedTypes: ImplicitAnnotatedAllowedTypes,
+): allowedTypes is AnnotatedAllowedTypes {
+	return (
+		typeof allowedTypes === "object" && "metadata" in allowedTypes && "types" in allowedTypes
```

Comment:
this check does not look safe (seems like a static named "metadata and one named types on a schema class would break it), but it is safer than it appears at first look.  A comment would help:

I realize this code was just moved, not modified in this PR, so if you want, fixing it in a different PR might make sense.

Anyway, isAnnotatedAllowedType should probably be implemented using `!isLazy`.

```suggestion
		// Class based schema, and lazy schema references report type "function": filtering them out with typeof makes narrowing based on members mostly safe
		typeof allowedTypes === "object" && "metadata" in allowedTypes && "types" in allowedTypes
```